src/ICeWall/doc/set_padring_options.md

```diff
@@ -0,0 +1,56 @@
+## set_padring_options
+### Synopsis
+```
+  % set_padring_options \
+      -type (flipchip|wirebond) \
+      [-power power_nets] \
+      [-ground ground_nets] \
+      [-offsets offsets] \
```

Comment:
When I only use a single number for the offsets, the corner cells end up with negative locations (except for the LL). If I write in "55 55 55 55", I get the correct pad ring, so maybe there is a bug here when only defining the number once?

Reply:
Strange - this doesnt happen with my testcases - is there something happening with a very small die_size being defined in the initialize_floorplan step?

Reply:
there might have been an issue in my setup, issue is gone. I think it was initialized to the wrong footprint when I was copying it over.

Reply:
@Colin-Holehouse I figured it out, it seems to have nothing to do with -offsets "55" vs "55 55 55 55". I am defining the library cells before initializing the floorplan. That seems to cause a problem. I tried moving the call to just after the first add_pad_libcell and it's still an issue. I can only get the corners if I initialize before defining any cells.

Using the tcl_interface.example.tcl and changing the order of the commands, you can get the same behavior:
# Load library defintions
add_pad_libcell \
  -name PAD \
  -orient {bottom R0 right R90 top R180 left R270} \
  -bondpad \
  -pad_pin_name PAD 

initialize_floorplan \
  -die_area  {0 0 3000.000 3000.000} \
  -core_area {180.012 180.096 2819.964 2819.712} \
  -site      FreePDK45_38x28_10R_NP_162NW_34O
make_tracks

source ../../../test/Nangate45/Nangate45.tracks

Reply:
@Colin-Holehouse I think I've tracked down where this error stems from.  When you check to see if the `$db` is initialized in the `add_pad` or `add_libcell` and you call `initialize`, this is setting the `chip_width` and `chip_height`. Which is `initialize_floorplan` hasn't been called, the height and width are zero.  Since these two parameters are not used anywhere during setup, you could reset them at the start of `init_footprint` where it looks like you call `initialize`, but the values do not appear to get updated, that should be easy to correct and then this issue should go away.

Reply:
Thanks for chasing this down, 
I updated the initialization so that define_pad_cell can be called after the library cells have been loaded, and it throws an error if the technology has not been loaded
The other commands can be run after the design has been loaded, even before the floorplan had been initialized - an error will be thrown if the design is not loaded